wsmaster/che-core-api-workspace-activity/src/main/java/org/eclipse/che/api/workspace/activity/WorkspaceActivityChecker.java

```diff
@@ -153,6 +160,29 @@ private void checkActivityRecordValidity() throws ServerException {
     }
   }
 
+  /**
+   * Makes sure that any activity records are rectified if they do not reflect the true state of the
+   * workspace anymore.
+   *
+   * @throws ServerException or error
+   */
+  private void reconcileActivityStatuses() throws ServerException {
+    for (WorkspaceActivity a : Pages.iterateLazily(activityDao::getAll, 200)) {
+      WorkspaceStatus status = workspaceRuntimes.getStatus(a.getWorkspaceId());
+      if (a.getStatus() != status) {
+        if (LOG.isWarnEnabled()) {
```

Comment:
I think you should not invoke this method... LOG.warn should just do not have any effect if the log level is greater than warn...
If I'm mistaken - should we check that same for all level? like IfErrorEnabled?

Reply:
This is here because `LOG.warn` with more than 2 parameters is actually `LOG.warn(String, Object...)` so the `isWarnEnabled()` check prevents the unnecessary instantiation and initialization of the array to pass to that method.

Reply:
It's so low level of optimization...
I would not do it only in one place. 
If it's really important - then let's solve this issue in general.
But it's up to you, feel free to leave as is.